agent/tests/PendingConnection.php

```diff
@@ -19,11 +19,8 @@ public function __invoke(): void
 
         $connection->emit('data', [$this->payload]);
 
-        $connection->emit('end');
-
         $connection->emit('close');
 
-        $connection->removeAllListeners();
```

Comment:
This is now handled in the agent-wrapper along with the same work we do to the server.